addons/mail/models/mail_channel.py

```diff
@@ -65,6 +66,7 @@ def _generate_random_token(self):
     channel_member_ids = fields.One2many(
         'mail.channel.member', 'channel_id', string='Members',
         groups='base.group_user')
+    pinned_message_ids = fields.One2many('mail.message', 'res_id', domain=lambda self: [('model', '=', self._name), ('pinned_at', '!=', False)], string='Pinned Messages')
```

Comment:
hardcode `mail.channel` instead of less obvious `self._name`?
But I'm not sure to understand what is the goal of that domain? If I'm understand it correctly, shouldn't it be also restricted to current channel id?

Reply:
The one2many is on `res_id` so without the domain the field would contain every message having `res_id` equal to `channel_id`. The domain then restrict those messages to 1 - the one having the appropriate model 2 - being pinned. The result is the list of pinned messages on the channel, I believe it is OK.